tex/chProgramming.tex

```diff
@@ -1310,8 +1321,8 @@ \section{Containers}\label{sec:poly}
 
 As expected, \C{listn} %elements of this data type
 cannot hold boolean values.
-So if we need to
-manipulate a list of booleans we have to define a similar data type:
+So, if we need to manipulate a list of booleans,
```

Comment:
I don't approve this change

Reply:
Grammatically, I believe, it's fine either way ( https://english.stackexchange.com/questions/4693/if-with-implicit-then-question-should-a-comma-be-used ); I personally think the sentence is easier to parse with the commas in it (otherwise, "a list of booleans we have" sounds like an object).

Reply:
My point is only to avoid too many commas in short succession: as a reader I see a comma as a place to take your breath, and if you take your breath too often, you choke (in other words, avoiding too many commas keeps some musicality in the phrasing). So, I would kept the comma if there was not already another one in the sentence.

Reply:
"if we need to manipulate a list of booleans" takes several seconds to pronounce; I wouldn't worry about my breathing there :)

Reply:
Although it's grammatically valid either way, I think not having the comma makes it flow better.